lib/Optimizer/Transforms/QuakeAddMetadata.cpp

```diff
@@ -122,6 +124,35 @@ struct QuakeFunctionAnalysis {
                 return WalkResult::interrupt();
               }
             }
+
+            // Look for any future cast/compute_ptr/load, and if that load is
+            // used by a conditional statement
+            if (auto cast = dyn_cast<cudaq::cc::CastOp>(allocUser)) {
+              for (auto castUser : cast->getUsers()) {
+                if (auto cp = dyn_cast<cudaq::cc::ComputePtrOp>(castUser)) {
+                  for (auto cpUser : cp->getUsers()) {
+                    if (auto load = dyn_cast<cudaq::cc::LoadOp>(cpUser)) {
+                      auto loadUser = *load->getUsers().begin();
+
+                      // Loaded Val could be used directly or by an Arith
+                      // boolean operation
+                      while (loadUser->getDialect()->getNamespace() ==
+                             "arith") {
+                        auto res = loadUser->getResult(0);
+                        loadUser = *res.getUsers().begin();
+                      }
+
+                      // At this point we should be able to check if we are
+                      // being used by a conditional
+                      if (isa<cudaq::cc::IfOp, cf::CondBranchOp>(loadUser)) {
+                        data.hasConditionalsOnMeasure = true;
+                        return WalkResult::interrupt();
+                      }
+                    }
+                  }
+                }
+              }
+            }
```

Comment:
> optional: could remove some of these braces

I think that removing them would go against this section of the [LLVM Coding Standards](https://llvm.org/docs/CodingStandards.html#don-t-use-braces-on-simple-single-statement-bodies-of-if-else-loop-statements):

```cpp
// Use braces for the outer `if` since the nested `for` is braced.
if (isa<FunctionDecl>(D)) {
  for (auto *A : D.attrs()) {
    // In this `for` loop body, it is necessary that we explain the situation
    // with this surprisingly long comment, forcing braces on the `for` block.
    handleAttr(A);
  }
}
```

It would also deviate from what was done in lines 104-126, so I think I'd rather leave them.